Code clean-up 03/29/25

sw23 opened the pull request:
Fixes #240 #238

FYI @aDesertRat

sw23 committed:
Code clean-up 03/29/25

aDesertRat commented:
Thank you. I was struggling with this. I suspected the checkbox flags were out of bounds but would have never guessed to bump the mask. I also missed that the created total is not always the number used in SharedTest.
It seems that if I have a PR all commits are added to it even when they have been pulled. I haven't been able to control what is in the PR. Sorry about the extra chatter there.

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

On Saturday, March 29th, 2025 at 1:12 PM, Spencer Williams ***@***.***> wrote:

> Merged [#242](https://github.com/anwilli5/coin-collection-android-US/pull/242) into main.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/anwilli5/coin-collection-android-US/pull/242#event-17046018835), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BJWPVSYHSPXD3JYE742L5TD2W35DXAVCNFSM6AAAAAB2BVJBICVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXGA2DMMBRHA4DGNI).
> You are receiving this because you were mentioned.Message ID: ***@***.***>

aDesertRat commented:
Took the app to my coin club and got some interesting feedback.
a) how do I create a already completed collection? (simple? or wish list ?)
b) how do I change the icon on the main collections list page? (wish list )
c) can I add my own icons? ( wish list )
d) your missing some roosevelt dimes. (I have fixed this - in PR)
e) some Ikes are missing mint marks. (I have fixed this- in PR)
f) Dollar icons need help ( I will address this and include it in my PR)
g) coin labels need help (I have addressed this- in PR)
h) coin order for westward nickels is messed up. (I have fixed this- in PR)
i) coin edit box needs a info entry to show larger front and back, mintage,
and notes edit box. ( wish list )

wish list to consider for next year?

Quite a discussion about images. I explained we could only distribute GPL'ed or public domain. They said as individuals they could pirate much better images. Import image function seems to be the solution.

I have looked at importing graphics and found code to use the file manager to navigate
to a file and select only PNG. I have found code to enforce the size (resize). I have found code to save the file. A text box to inform the user is not hard. How to get it into or dellete it from the imageId's remains a mystery.........coming attractions for 2026 release.

I note that it is possible to divorce code and coin data by creating collections using list filtering and lists can be made from the export function. I can also edit lists to include mintages. If I also copy how you created imageId's to create revimageId's the info box is doable. I may try a prototype class in the coming months. What say you.

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

On Sunday, March 30th, 2025 at 8:36 AM, adesertrat ***@***.***> wrote:

> Thank you. I was struggling with this. I suspected the checkbox flags were out of bounds but would have never guessed to bump the mask. I also missed that the created total is not always the number used in SharedTest.
> It seems that if I have a PR all commits are added to it even when they have been pulled. I haven't been able to control what is in the PR. Sorry about the extra chatter there.
>
> Sent with [Proton Mail](https://proton.me/mail/home) secure email.
>
> On Saturday, March 29th, 2025 at 1:12 PM, Spencer Williams ***@***.***> wrote:
>
>> Merged [#242](https://github.com/anwilli5/coin-collection-android-US/pull/242) into main.
>>
>> —
>> Reply to this email directly, [view it on GitHub](https://github.com/anwilli5/coin-collection-android-US/pull/242#event-17046018835), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BJWPVSYHSPXD3JYE742L5TD2W35DXAVCNFSM6AAAAAB2BVJBICVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXGA2DMMBRHA4DGNI).
>> You are receiving this because you were mentioned.Message ID: ***@***.***>

sw23 commented:
> Thank you. I was struggling with this. I suspected the checkbox flags were out of bounds but would have never guessed to bump the mask. I also missed that the created total is not always the number used in SharedTest. It seems that if I have a PR all commits are added to it even when they have been pulled. I haven't been able to control what is in the PR. Sorry about the extra chatter there. Sent with [Proton Mail](https://proton.me/mail/home) secure email.
> […](#)
> On Saturday, March 29th, 2025 at 1:12 PM, Spencer Williams ***@***.***> wrote: Merged [#242](#242) into main. — Reply to this email directly, [view it on GitHub]([#242 (comment)](https://github.com/anwilli5/coin-collection-android-US/pull/242#event-17046018835)), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BJWPVSYHSPXD3JYE742L5TD2W35DXAVCNFSM6AAAAAB2BVJBICVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXGA2DMMBRHA4DGNI). You are receiving this because you were mentioned.Message ID: ***@***.***>

@aDesertRat - the PRs are tied to the branch that you are on. By default, you will be on your cloned repo's 'main' branch, but you can create other branches as well for different features. To create a branch, run `git checkout -b my_new_branch`. You can then `git add` and `git commit` like you normally would, and then make a new PR from the branch into this repo's main branch. It's sometimes handy to have multiple branches going at a time. To switch back to the main branch, run `git checkout main`. You can see what branches you have by running `git branch`. Note that branches will be local to your machine until you run `git push`, similar to changes that you've committed to your main branch

sw23 commented:
> Took the app to my coin club and got some interesting feedback. a) how do I create a already completed collection? (simple? or wish list ?) b) how do I change the icon on the main collections list page? (wish list ) c) can I add my own icons? ( wish list ) d) your missing some roosevelt dimes. (I have fixed this - in PR) e) some Ikes are missing mint marks. (I have fixed this- in PR) f) Dollar icons need help ( I will address this and include it in my PR) g) coin labels need help (I have addressed this- in PR) h) coin order for westward nickels is messed up. (I have fixed this- in PR) i) coin edit box needs a info entry to show larger front and back, mintage, and notes edit box. ( wish list ) wish list to consider for next year? Quite a discussion about images. I explained we could only distribute GPL'ed or public domain. They said as individuals they could pirate much better images. Import image function seems to be the solution. I have looked at importing graphics and found code to use the file manager to navigate to a file and select only PNG. I have found code to enforce the size (resize). I have found code to save the file. A text box to inform the user is not hard. How to get it into or dellete it from the imageId's remains a mystery.........coming attractions for 2026 release. I note that it is possible to divorce code and coin data by creating collections using list filtering and lists can be made from the export function. I can also edit lists to include mintages. If I also copy how you created imageId's to create revimageId's the info box is doable. I may try a prototype class in the coming months. What say you. Sent with [Proton Mail](https://proton.me/mail/home) secure email.
> […](#)
> On Sunday, March 30th, 2025 at 8:36 AM, adesertrat ***@***.***> wrote: Thank you. I was struggling with this. I suspected the checkbox flags were out of bounds but would have never guessed to bump the mask. I also missed that the created total is not always the number used in SharedTest. It seems that if I have a PR all commits are added to it even when they have been pulled. I haven't been able to control what is in the PR. Sorry about the extra chatter there. Sent with [Proton Mail](https://proton.me/mail/home) secure email. On Saturday, March 29th, 2025 at 1:12 PM, Spencer Williams ***@***.***> wrote: > Merged [#242](#242) into main. > > — > Reply to this email directly, [view it on GitHub]([#242 (comment)](https://github.com/anwilli5/coin-collection-android-US/pull/242#event-17046018835)), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BJWPVSYHSPXD3JYE742L5TD2W35DXAVCNFSM6AAAAAB2BVJBICVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXGA2DMMBRHA4DGNI). > You are receiving this because you were mentioned.Message ID: ***@***.***>

Nice feedback! Adding some comments below

a) how do I create a already completed collection? (simple? or wish list ?)
 - [SW] Does this mean they want an easy way to mark all coins as collected? That would be an easy button to add to the three dots list on the coin page.

b) how do I change the icon on the main collections list page? (wish list )
- [SW] Interesting idea - opened this work item for tracking
- #246

c) can I add my own icons? ( wish list )
- [SW] I like this idea as well
- #20

d) your missing some roosevelt dimes. (I have fixed this - in PR) 👍
e) some Ikes are missing mint marks. (I have fixed this- in PR) 👍
f) Dollar icons need help ( I will address this and include it in my PR) 👍
g) coin labels need help (I have addressed this- in PR) 👍
h) coin order for westward nickels is messed up. (I have fixed this- in PR) 👍
i) coin edit box needs a info entry to show larger front and back, mintage,
and notes edit box. ( wish list )
- [SW] Is this part of what they want?
- #25 
- [SW] Is making the images larger part of their request also? For the mintage/notes, is that a change to the advanced view that they want?


I note that it is possible to divorce code and coin data by creating collections using list filtering and lists can be made from the export function. I can also edit lists to include mintages.
[SW] I have thought about doing this as well, but haven't gotten around to it. It's a good idea though. I'll open a work item for tracking
- #247

I have looked at importing graphics and found code to use the file manager to navigate
to a file and select only PNG. I have found code to enforce the size (resize). I have found code to save the file. A text box to inform the user is not hard. How to get it into or dellete it from the imageId's remains a mystery.........coming attractions for 2026 release.
[SW] I am considering adding this in the coming months. I don't think it will be that hard - I have GitHub Copilot now which should make adding it much faster.

If I also copy how you created imageId's to create revimageId's the info box is doable. I may try a prototype class in the coming months. What say you.
[SW] Feel free to give it a shot! As of now I haven't thought too much about what having both sides would look like. I guess I use my imagination but I know some people would like to see both sides

aDesertRat commented:
I know you are busy trying to get out an update.

For a) if its easy to lite all coins with a entry in the three dot menu, cool.
For i) i think they wanted information to inform purchases. If they just want a reverse they can add one for many series out of the imageId spinbox like some coin folios have a space for a reverse. It also would not be hard to offer a lead off reverse as a option. May happen for 2026 update. If fully implemented does that close #25?

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

On Thursday, April 3rd, 2025 at 10:17 PM, Spencer Williams ***@***.***> wrote:

>> Took the app to my coin club and got some interesting feedback. a) how do I create a already completed collection? (simple? or wish list ?) b) how do I change the icon on the main collections list page? (wish list ) c) can I add my own icons? ( wish list ) d) your missing some roosevelt dimes. (I have fixed this - in PR) e) some Ikes are missing mint marks. (I have fixed this- in PR) f) Dollar icons need help ( I will address this and include it in my PR) g) coin labels need help (I have addressed this- in PR) h) coin order for westward nickels is messed up. (I have fixed this- in PR) i) coin edit box needs a info entry to show larger front and back, mintage, and notes edit box. ( wish list ) wish list to consider for next year? Quite a discussion about images. I explained we could only distribute GPL'ed or public domain. They said as individuals they could pirate much better images. Import image function seems to be the solution. I have looked at importing graphics and found code to use the file manager to navigate to a file and select only PNG. I have found code to enforce the size (resize). I have found code to save the file. A text box to inform the user is not hard. How to get it into or dellete it from the imageId's remains a mystery.........coming attractions for 2026 release. I note that it is possible to divorce code and coin data by creating collections using list filtering and lists can be made from the export function. I can also edit lists to include mintages. If I also copy how you created imageId's to create revimageId's the info box is doable. I may try a prototype class in the coming months. What say you. Sent with [Proton Mail](https://proton.me/mail/home) secure email.
>> […](#)
>> On Sunday, March 30th, 2025 at 8:36 AM, adesertrat @.> wrote: Thank you. I was struggling with this. I suspected the checkbox flags were out of bounds but would have never guessed to bump the mask. I also missed that the created total is not always the number used in SharedTest. It seems that if I have a PR all commits are added to it even when they have been pulled. I haven't been able to control what is in the PR. Sorry about the extra chatter there. Sent with [Proton Mail](https://proton.me/mail/home) secure email. On Saturday, March 29th, 2025 at 1:12 PM, Spencer Williams @.> wrote: > Merged #242 into main. > > — > Reply to this email directly, [view it on GitHub]([#242 (comment)](https://github.com/anwilli5/coin-collection-android-US/pull/242#event-17046018835)), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BJWPVSYHSPXD3JYE742L5TD2W35DXAVCNFSM6AAAAAB2BVJBICVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXGA2DMMBRHA4DGNI). > You are receiving this because you were mentioned.Message ID: @.***>
>
> Nice feedback! Adding some comments below
>
> a) how do I create a already completed collection? (simple? or wish list ?)
>
> - [SW] Does this mean they want an easy way to mark all coins as collected? That would be an easy button to add to the three dots list on the coin page.
>
> b) how do I change the icon on the main collections list page? (wish list )
>
> - [SW] Interesting idea - opened this work item for tracking
> - [#246](https://github.com/anwilli5/coin-collection-android-US/issues/246)
>
> c) can I add my own icons? ( wish list )
>
> - [SW] I like this idea as well
> - [#20](https://github.com/anwilli5/coin-collection-android-US/issues/20)
>
> d) your missing some roosevelt dimes. (I have fixed this - in PR) 👍
> e) some Ikes are missing mint marks. (I have fixed this- in PR) 👍
> f) Dollar icons need help ( I will address this and include it in my PR) 👍
> g) coin labels need help (I have addressed this- in PR) 👍
> h) coin order for westward nickels is messed up. (I have fixed this- in PR) 👍
> i) coin edit box needs a info entry to show larger front and back, mintage,
> and notes edit box. ( wish list )
>
> - [SW] Is this part of what they want?
> - [#25](https://github.com/anwilli5/coin-collection-android-US/issues/25)
> - [SW] Is making the images larger part of their request also? For the mintage/notes, is that a change to the advanced view that they want?
>
> I note that it is possible to divorce code and coin data by creating collections using list filtering and lists can be made from the export function. I can also edit lists to include mintages.
> [SW] I have thought about doing this as well, but haven't gotten around to it. It's a good idea though. I'll open a work item for tracking
>
> - [#247](https://github.com/anwilli5/coin-collection-android-US/issues/247)
>
> I have looked at importing graphics and found code to use the file manager to navigate
> to a file and select only PNG. I have found code to enforce the size (resize). I have found code to save the file. A text box to inform the user is not hard. How to get it into or dellete it from the imageId's remains a mystery.........coming attractions for 2026 release.
> [SW] I am considering adding this in the coming months. I don't think it will be that hard - I have GitHub Copilot now which should make adding it much faster.
>
> If I also copy how you created imageId's to create revimageId's the info box is doable. I may try a prototype class in the coming months. What say you.
> [SW] Feel free to give it a shot! As of now I haven't thought too much about what having both sides would look like. I guess I use my imagination but I know some people would like to see both sides
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/anwilli5/coin-collection-android-US/pull/242#issuecomment-2777579913), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BJWPVS2PB5D7ABCXROZVY5T2XYIVRAVCNFSM6AAAAAB2BVJBICVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZXGU3TSOJRGM).
> You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> [sw23]sw23 left a comment [(anwilli5/coin-collection-android-US#242)](https://github.com/anwilli5/coin-collection-android-US/pull/242#issuecomment-2777579913)
>
>> Took the app to my coin club and got some interesting feedback. a) how do I create a already completed collection? (simple? or wish list ?) b) how do I change the icon on the main collections list page? (wish list ) c) can I add my own icons? ( wish list ) d) your missing some roosevelt dimes. (I have fixed this - in PR) e) some Ikes are missing mint marks. (I have fixed this- in PR) f) Dollar icons need help ( I will address this and include it in my PR) g) coin labels need help (I have addressed this- in PR) h) coin order for westward nickels is messed up. (I have fixed this- in PR) i) coin edit box needs a info entry to show larger front and back, mintage, and notes edit box. ( wish list ) wish list to consider for next year? Quite a discussion about images. I explained we could only distribute GPL'ed or public domain. They said as individuals they could pirate much better images. Import image function seems to be the solution. I have looked at importing graphics and found code to use the file manager to navigate to a file and select only PNG. I have found code to enforce the size (resize). I have found code to save the file. A text box to inform the user is not hard. How to get it into or dellete it from the imageId's remains a mystery.........coming attractions for 2026 release. I note that it is possible to divorce code and coin data by creating collections using list filtering and lists can be made from the export function. I can also edit lists to include mintages. If I also copy how you created imageId's to create revimageId's the info box is doable. I may try a prototype class in the coming months. What say you. Sent with [Proton Mail](https://proton.me/mail/home) secure email.
>> […](#)
>> On Sunday, March 30th, 2025 at 8:36 AM, adesertrat @.> wrote: Thank you. I was struggling with this. I suspected the checkbox flags were out of bounds but would have never guessed to bump the mask. I also missed that the created total is not always the number used in SharedTest. It seems that if I have a PR all commits are added to it even when they have been pulled. I haven't been able to control what is in the PR. Sorry about the extra chatter there. Sent with [Proton Mail](https://proton.me/mail/home) secure email. On Saturday, March 29th, 2025 at 1:12 PM, Spencer Williams @.> wrote: > Merged #242 into main. > > — > Reply to this email directly, [view it on GitHub]([#242 (comment)](https://github.com/anwilli5/coin-collection-android-US/pull/242#event-17046018835)), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BJWPVSYHSPXD3JYE742L5TD2W35DXAVCNFSM6AAAAAB2BVJBICVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJXGA2DMMBRHA4DGNI). > You are receiving this because you were mentioned.Message ID: @.***>
>
> Nice feedback! Adding some comments below
>
> a) how do I create a already completed collection? (simple? or wish list ?)
>
> - [SW] Does this mean they want an easy way to mark all coins as collected? That would be an easy button to add to the three dots list on the coin page.
>
> b) how do I change the icon on the main collections list page? (wish list )
>
> - [SW] Interesting idea - opened this work item for tracking
> - [#246](https://github.com/anwilli5/coin-collection-android-US/issues/246)
>
> c) can I add my own icons? ( wish list )
>
> - [SW] I like this idea as well
> - [#20](https://github.com/anwilli5/coin-collection-android-US/issues/20)
>
> d) your missing some roosevelt dimes. (I have fixed this - in PR) 👍
> e) some Ikes are missing mint marks. (I have fixed this- in PR) 👍
> f) Dollar icons need help ( I will address this and include it in my PR) 👍
> g) coin labels need help (I have addressed this- in PR) 👍
> h) coin order for westward nickels is messed up. (I have fixed this- in PR) 👍
> i) coin edit box needs a info entry to show larger front and back, mintage,
> and notes edit box. ( wish list )
>
> - [SW] Is this part of what they want?
> - [#25](https://github.com/anwilli5/coin-collection-android-US/issues/25)
> - [SW] Is making the images larger part of their request also? For the mintage/notes, is that a change to the advanced view that they want?
>
> I note that it is possible to divorce code and coin data by creating collections using list filtering and lists can be made from the export function. I can also edit lists to include mintages.
> [SW] I have thought about doing this as well, but haven't gotten around to it. It's a good idea though. I'll open a work item for tracking
>
> - [#247](https://github.com/anwilli5/coin-collection-android-US/issues/247)
>
> I have looked at importing graphics and found code to use the file manager to navigate
> to a file and select only PNG. I have found code to enforce the size (resize). I have found code to save the file. A text box to inform the user is not hard. How to get it into or dellete it from the imageId's remains a mystery.........coming attractions for 2026 release.
> [SW] I am considering adding this in the coming months. I don't think it will be that hard - I have GitHub Copilot now which should make adding it much faster.
>
> If I also copy how you created imageId's to create revimageId's the info box is doable. I may try a prototype class in the coming months. What say you.
> [SW] Feel free to give it a shot! As of now I haven't thought too much about what having both sides would look like. I guess I use my imagination but I know some people would like to see both sides
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/anwilli5/coin-collection-android-US/pull/242#issuecomment-2777579913), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BJWPVS2PB5D7ABCXROZVY5T2XYIVRAVCNFSM6AAAAAB2BVJBICVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONZXGU3TSOJRGM).
> You are receiving this because you were mentioned.Message ID: ***@***.***>
